graphite_beacon/handlers/telegram.py

```diff
@@ -23,89 +44,173 @@ def init_handler(self):
-                                "reply_to_message_id": update["message"]["message_id"],
-                                "text": "This chat is already activated."}),
-                            method="POST",
-                            headers={"Content-Type": "application/json"})
-                except:
-                    continue
+
+        data = get_data(upd, self.bot_ident)
+        for (update_id, chat_id, message_id, command) in data:
+            self._last_update = update_id
+            chat_is_known = (chat_id in chats)
+            reply_text = None
+
+            if command == '/activate':
+                if chat_is_known:
+                    reply_text = 'This chat is already activated.'
+                else:
+                    LOGGER.debug(
+                            'Adding chat [%s] to notify list.', chat_id)
+                    reply_text = 'Activated.'
+                    chats.add(chat_id)
+                    if chatfile:
+                        write_to_file(chats, chatfile)
+
+            elif command == '/deactivate':
+                if chat_is_known:
+                    LOGGER.debug(
+                            'Deleting chat [%s] from notify list.', chat_id)
+                    reply_text = 'Deactivated.'
+                    chats.remove(chat_id)
+                    if chatfile:
+                        write_to_file(chats, chatfile)
+
+            elif command == '/help':
+                reply_text = HELP_MESSAGE
+
             else:
-                continue
+                LOGGER.warning(BOT_COMMAND_ERRORS[command])
+
+            if reply_text:
+                yield self.sendMessage({
+                        'chat_id': chat_id,
+                        'reply_to_message_id': message_id,
+                        'text': reply_text,})
 
     @gen.coroutine
     def notify(self, level, *args, **kwargs):
 
-        LOGGER.debug("Handler (%s) %s", self.name, level)
+        LOGGER.debug('Handler (%s) %s', self.name, level)
+
+        notify_text = self.get_message(level, *args, **kwargs)
+        for chat in self.chats:
+            yield self.sendMessage({"chat_id": chat, "text": notify_text})
 
-        message = self.get_message(level, *args, **kwargs)
-        for chat in self._chats:
-            yield self.client.fetch(
-                self.url + "sendMessage", body=json.dumps({"chat_id": chat, "text": message}),
-                method="POST", headers={"Content-Type": "application/json"})
+    def get_message(self, level, alert, value,
+                    target=None, ntype=None, rule=None):
+
+        msg_type = 'telegram' if ntype == 'graphite' else 'short'
+        tmpl = TEMPLATES[ntype][msg_type]
+        generated = tmpl.generate(
+            level=level, reactor=self.reactor, alert=alert,
+            value=value, target=target,)
+        return generated.decode().strip()
+
+def write_to_file(chats, chatfile):
+    with open(chatfile, 'w') as fh:
+        chats = map(str, chats)
```

Comment:
What's the utility of this? Looks like it just turns a set into a list, but sets are iterable so this shouldn't be necessary